sklearn/utils/testing.py

```diff
@@ -714,26 +714,14 @@ def set_random_state(estimator, random_state=0):
         estimator.set_params(random_state=random_state)
 
 
-def if_matplotlib(func):
-    """Test decorator that skips test if matplotlib not installed.
-
-    Parameters
-    ----------
-    func
-    """
-    @wraps(func)
-    def run_test(*args, **kwargs):
-        try:
-            import matplotlib
-            matplotlib.use('Agg', warn=False)
-            # this fails if no $DISPLAY specified
-            import matplotlib.pyplot as plt
-            plt.figure()
-        except ImportError:
-            raise SkipTest('Matplotlib not available.')
-        else:
-            return func(*args, **kwargs)
-    return run_test
+def has_matplotlib():
+    """Check if matplotlib is installed for testing purpose."""
+    try:
+        import matplotlib  # noqa
+        matplotlib.use('Agg', warn=False)
```

Comment:
Why do we need this line?

Reply:
For the test using `matplotlib==1.5.1`, the backend is not set by default. It was one of the earlier failure.